fn.go

```diff
@@ -24,15 +38,227 @@ func (f *Function) RunFunction(_ context.Context, req *fnv1beta1.RunFunctionRequ
 
 	rsp := response.To(req, response.DefaultTTL)
 
+	// Get function input.
 	in := &v1beta1.Input{}
 	if err := request.GetInput(req, in); err != nil {
 		response.Fatal(rsp, errors.Wrapf(err, "cannot get Function input from %T", req))
 		return rsp, nil
 	}
 
-	// TODO: Add your Function logic here!
-	response.Normalf(rsp, "I was run with input %q!", in.Example)
-	f.log.Info("I was run!", "input", in.Example)
+	// Get XR the pipeline targets.
+	oxr, err := request.GetObservedCompositeResource(req)
+	if err != nil {
+		response.Fatal(rsp, errors.Wrapf(err, "cannot get observed composite resource"))
+		return rsp, nil
+	}
+	f.log.Debug("asdf", "bsdf", oxr)
+
+	// Build extraResource Requests.
+	requirements, err := buildRequirements(in, oxr)
+	if err != nil {
+		response.Fatal(rsp, errors.Wrapf(err, "could not build extraResource requirements"))
+		return rsp, nil
+	}
+	rsp.Requirements = requirements
+
+	// Verify that extra resources were even requested in input.
+	// Appears 100% needed.
```

Comment:
For some reason even when I know my function request has `extraResources` fully speced I still get an issue here.

Don't understand. It's like a blank request is sent on startup?